components/webext-storage/src/db.rs

```diff
@@ -69,24 +69,16 @@ impl StorageDb {
 
     /// Closes the database connection. If there are any unfinalized prepared
     /// statements on the connection, `close` will fail and the `StorageDb` will
-    /// be returned to the caller so that it can retry, drop (via `mem::drop`)
-    // or leak (`mem::forget`) the connection.
-    ///
-    /// Keep in mind that dropping the connection tries to close it again, and
-    /// panics on error.
-    pub fn close(self) -> result::Result<(), (StorageDb, Error)> {
-        let StorageDb {
-            writer,
-            interrupt_handle,
-        } = self;
-        writer.close().map_err(|(writer, err)| {
-            (
-                StorageDb {
-                    writer,
-                    interrupt_handle,
-                },
-                err.into(),
-            )
+    /// remain open and the connection will be leaked - we used to return the
+    /// underlying connection so the caller can retry but (a) that's very tricky
+    /// in an Arc<Mutex<>> world and (b) we never actually took advantage of
+    /// that retry capability.
+    pub fn close(self) -> Result<()> {
+        self.writer.close().map_err(|(writer, err)| {
+            // If we just let `writer` drop, the close would panic on failure.
+            // (Actually, not sure that's true? Regardless, this seems fine)
```

Comment:
I think [this is where the panic happens](https://github.com/rusqlite/rusqlite/blob/9880cdef12c1b624e351ed077dd71a2f77978f77/src/inner_connection.rs#L380), but maybe that's wrong?

Reply:
I think that is completely correct, thanks! I'll update the comment